plugins/chat/spec/system/create_channel_spec.rb

```diff
@@ -72,45 +78,75 @@
       fab!(:private_category_1) { Fabricate(:private_category, group: group_1) }
 
       it "escapes the group name" do
-        visit("/chat")
-        find(".new-channel-btn").click
-        find(".category-chooser").click
-        find(".category-row[data-value=\"#{private_category_1.id}\"]").click
+        chat_page.visit_browse
+        chat_page.new_channel_button.click
+        channel_modal.select_category(private_category_1)
 
-        expect(find(".create-channel-hint")["innerHTML"].strip).to include(
+        expect(channel_modal.create_channel_hint["innerHTML"].strip).to include(
           "&lt;script&gt;e&lt;/script&gt;",
         )
       end
     end
 
+    it "autogenerates slug from name and changes slug placeholder" do
+      chat_page.visit_browse
+      chat_page.new_channel_button.click
+      name = "Cats & Dogs"
+      channel_modal.select_category(category_1)
+      channel_modal.fill_name(name)
+      channel_modal.fill_description("All kind of cute cats")
+
+      wait_for_attribute(channel_modal.slug_input, :placeholder, "cats-dogs")
```

Comment:
Looking for an opinion on this, not sure if it's crazy to do but capybara does not have an inbuilt way to wait for an attribute value.